server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -990,11 +1020,20 @@ public long mappingVersion() {
             return mappingVersion;
         }
 
+        public long settingsVersion() {
+            return settingsVersion;
+        }
+
         public Builder mappingVersion(final long mappingVersion) {
             this.mappingVersion = mappingVersion;
             return this;
         }
 
+        public Builder settingsVersion(final long settingsVersion) {
+            this.settingsVersion = settingsVersion;
```

Comment:
would it make sense to check that we always increase?

Reply:
That would not be correct since this method is called when building index metadata from parsing or de-serializing.